cmd/charm/charmcmd/cmd.go

```diff
@@ -138,24 +139,56 @@ func newCharmStoreClient(ctxt *cmd.Context, auth authInfo) (*csClient, error) {
 			BakeryClient: bakeryClient,
 			User:         auth.username,
 			Password:     auth.password,
-		}),
+		}).WithChannel(channel),
 		jar:  jar,
 		ctxt: ctxt,
 	}
 	return &csClient, nil
 }
 
+// addChannelFlag adds the -c (--channel) flags to the given flag set.
+func addChannelFlag(f *gnuflag.FlagSet, cv *chanValue) {
+	chans := []string{
+		string(params.StableChannel),
+		string(params.CandidateChannel),
+		string(params.BetaChannel),
+		string(params.EdgeChannel),
+		string(params.UnpublishedChannel),
+	}
+	f.Var(cv, "c", fmt.Sprintf("the channel the charm or bundle is assigned to (%s)", strings.Join(chans, "|")))
+	f.Var(cv, "channel", "")
+}
+
+// chanValue is a gnuflag.Value that stores a channel name
+// ("stable", "candidate", "beta", "edge" or "unpublished").
+type chanValue string
```

Comment:
I think I'd be tempted to define this as:

    type chanValue struct {
          C params.Channel
    }

Then you won't need the Channel method,
and Set becomes a little simpler.